opty/tests/test_direct_collocation.py

```diff
@@ -1907,3 +1907,405 @@ def test_attributes_read_only():
+        return grad
+
+    instance_constraints = (
+        x.func(t0) - 3.0,
+        y.func(t0) - 2.0,
+        ux.func(t0) - 5.0,
+
+        x.func(t1) - 1.5*b1,
+        y.func(t1) - 4.0,
+        ux.func(t1) - 1.0+b2,
+        uy.func(t1) - 4.5,
+
+        x.func(t2) + 2.0,
+        y.func(t2) - 3.0,
+
+        x.func(tf) + 1.0,
+        ux.func(tf) + 5.0,
+    )
+
+    bounds= {
+        a1: (-1.0, 10.0),
+        a2: (-1.0, 12.0),
+
+        u1: (-10.0, 1.0),
+        u2: (-1.0, 10.0),
+
+        x: (-5.0, 5.0 ),
+        ux: (-1.0, 1.0),
+        y: (-4.0, 4.0),
+        uy: (-1.0, 1.0),
+    }
+
+    prob = Problem(
+        obj,
+        obj_grad,
+        eom,
+        state_symbols,
+        num_nodes,
+        interval_value,
+        known_parameter_map=par_map,
+        instance_constraints=instance_constraints,
+        bounds=bounds,
+        time_symbol=t,
+    )
+
+    # Set the expected initial guess
+    expected_guess = np.zeros(prob.num_free)
+    duration = (num_nodes-1)*interval_value
+    # x - guess
+    start = round(t0/duration*num_nodes)
+    ende = round(t1/duration*num_nodes)
+    werte = np.linspace(3.0, 1.5*par_map[b1], ende-start)
+    expected_guess[0*num_nodes+start:0*num_nodes+ende] = werte
+    start = round(t1/duration*num_nodes)
+    ende = round(t2/duration*num_nodes)
+    werte = np.linspace(1.5*par_map[b1], -2.0, ende-start)
+    expected_guess[0*num_nodes+start:0*num_nodes+ende] = werte
+    start = round(t2/duration*num_nodes)
+    ende = round(tf/duration*num_nodes)
+    werte = np.linspace(-2.0, -1.0, ende-start)
+    expected_guess[0*num_nodes+start:0*num_nodes+ende] = werte
+     # y - guess
+    start = round(t0/duration*num_nodes)
+    ende = round(t1/duration*num_nodes)
+    werte = np.linspace(2.0, 4.0, ende-start)
+    expected_guess[1*num_nodes+start:1*num_nodes+ende] = werte
+    start = round(t1/duration*num_nodes)
+    ende = round(t2/duration*num_nodes)
+    werte = np.linspace(4.0, 3.0, ende-start)
+    expected_guess[1*num_nodes+start:1*num_nodes+ende] = werte
+    # ux - guess
+    start = round(t0/duration*num_nodes)
+    ende = round(t1/duration*num_nodes)
+    werte = np.linspace(5.0, 1.0-par_map[b2], ende-start)
+    expected_guess[2*num_nodes+start:2*num_nodes+ende] = werte
+    start = round(t1/duration*num_nodes)
+    ende = round(tf/duration*num_nodes)
+    werte = np.linspace(1.0-par_map[b2], -5.0, ende-start)
+    expected_guess[2*num_nodes+start:2*num_nodes+ende] = werte
+    # uy - guess
+    expected_guess[3*num_nodes:4*num_nodes] = 4.5
+    # u1 - guess
+    expected_guess[4*num_nodes:5*num_nodes] = -9.0/2.0
+    # u2 - guess
+    expected_guess[5*num_nodes:6*num_nodes] = 9.0/2.0
+    # a1 - guess
+    expected_guess[6*num_nodes] = 9/2
+    # a2 - guess
+    expected_guess[6*num_nodes+1] = 11.0/2.0
+
+
+    initial_guess = prob.create_linear_initial_guess()
+    np.testing.assert_allclose(initial_guess, expected_guess)
```

Comment:
Plot of first guess:

![image](https://github.com/user-attachments/assets/0344155a-2145-47ce-a2e1-d6e2fe30e983)